internal/cmd/gocdk/init_test.go

```diff
@@ -147,6 +148,69 @@ func TestInit(t *testing.T) {
 		}
 	})
 }
+func TestInferModulePath(t *testing.T) {
+	testCases := []struct {
+		name       string
+		testGOPATH func(string) string
+		wantErr    bool
+	}{
+		{
+			"no GOPATH entry",
+			func(dir string) string {
+				return ""
+			},
+			true,
+		},
+		{
+			"single GOPATH entry",
+			func(dir string) string {
+				return "GOPATH=" + dir
+			},
+			false,
+		},
+		{
+			"multiple GOPATH entries",
+			func(dir string) string {
+				multiPath := "GOPATH=" + dir + string(filepath.ListSeparator) + filepath.Join("", "some", "other", "path")
+				fmt.Println("INSIDE--" + multiPath)
+				return multiPath
+			},
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			dir, err := ioutil.TempDir("", testTempDirPrefix)
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer func() {
+				if err := os.RemoveAll(dir); err != nil {
+					t.Error(err)
+				}
+			}()
+
+			ctx := context.Background()
+			pctx := newTestProcessContext(dir)
+			pctx.env = []string{tc.testGOPATH(dir)}
```

Comment:
Instead of clobbering the existing `env`, try appending to it:
`pctx.env = append(pctx.env, tc.testGOPATH(dir))`

Reply:
for "systematically eliminating possibilities" purposes, this comment is to note that this results in exactly the same errors.